pkg/readers/adapters/amqp.go

```diff
@@ -24,12 +25,21 @@ func (a *Amqp) Listen(msgChannel chan<- messages.Message) {
 }
 
 func (a *Amqp) processMessages(writeChannel chan<- messages.Message) {
-	msg := messages.Message{}
-	for d := range a.ConsumedMessages {
-		msg.Id = d.DeliveryTag
-		msg.Body = d.Body
-
-		writeChannel <- msg
+	for {
+		select {
+		case err := <-a.Notify:
+			if err != nil {
+				klog.Fatalf("Error in connection: %s", err)
```

Comment:
If error, then exit (rely on infrastructure for exponential backoff, instead of implementing complex logic).

If not, log info and carry on, probably we're handling graceful shutdown